src/main/java/de/metas/ui/web/letter/LetterRestController.java

```diff
@@ -134,10 +149,53 @@ public JSONLetter getLetter(@PathVariable("letterId") final String letterId)
 		return JSONLetter.of(letter);
 	}
 
+	@Builder(builderMethodName = "fromLetterBuilder")
+	private I_C_Letter createPersistentLetter(final String subject, final String content)
+	{
+		final I_C_Letter persistentLetter = InterfaceWrapperHelper.newInstance(I_C_Letter.class);
+		persistentLetter.setLetterSubject(subject);
+		persistentLetter.setLetterBody(content);
+		persistentLetter.setLetterBodyParsed(content);
+		InterfaceWrapperHelper.save(persistentLetter);
+		return persistentLetter;
+	}
+
+	private I_C_Letter updatePersistentLetter(final WebuiLetter letter)
+	{
+		Check.assume(letter.getPersistentLetterId() > 0, "Letter ID should be > 0");
+		final Properties ctx = Env.getCtx();
+		final int C_BPartner_ID = Env.getContextAsInt(ctx, I_C_Letter.COLUMNNAME_C_BPartner_ID);
+		final int C_BPartner_Location_ID = Env.getContextAsInt(ctx, I_C_Letter.COLUMNNAME_C_BPartner_Location_ID);
+		final String bpartnerAddress = Env.getContext(ctx, I_C_Letter.COLUMNNAME_BPartnerAddress);
+		
+		final I_C_Letter persistentLetter = InterfaceWrapperHelper.create(ctx, letter.getPersistentLetterId(), I_C_Letter.class, ITrx.TRXNAME_ThreadInherited);
+		persistentLetter.setLetterSubject(letter.getSubject());
+		// field is mandatory
+		persistentLetter.setLetterBody(Joiner.on(" ").skipNulls().join(Arrays.asList(letter.getContent())));
+		persistentLetter.setLetterBodyParsed(letter.getContent());
+		persistentLetter.setC_BPartner_ID(C_BPartner_ID);
+		persistentLetter.setC_BPartner_Location_ID(C_BPartner_Location_ID);
+		persistentLetter.setBPartnerAddress(bpartnerAddress);
+		InterfaceWrapperHelper.save(persistentLetter);
+		return persistentLetter;
+	}
+
 	private File createPDFFile(final WebuiLetter letter)
 	{
+		final I_C_Letter persistentLetter = updatePersistentLetter(letter);
+		byte[] pdf = Services.get(ITextTemplateBL.class).createPDF(persistentLetter);
 		final String pdfFilenamePrefix = Services.get(IMsgBL.class).getMsg(Env.getCtx(), Letters.MSG_Letter);
-		final File pdfFile = MADBoilerPlate.getPDF(pdfFilenamePrefix, letter.getContent(), BoilerPlateContext.EMPTY);
+		File pdfFile = null;
+		try
+		{
+			pdfFile = File.createTempFile(pdfFilenamePrefix, ".pdf");
+			FileUtils.writeByteArrayToFile(pdfFile, pdf);
```

Comment:
i think u shall return the byte[] instead of saving a File.
Check the call trace and u will see that at the end byte[] is needed

Reply:
Yes, but for the request, we need the file also. But we can create it only then